common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -556,4 +556,18 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
     protected String keyToString(@k@ key) {
         return @O@.toString(key);
     }
+
+    /**
+     * Fast method of finding the next power of 2 greater than or equal to the supplied value.
+     *
+     * If the value is &lt;= 0 then 1 will be returned.
+     *
+     * This method is not suitable for {@link Integer#MIN_VALUE} or numbers greater than 2^30.*
+     * @param value from which to search for next power of 2
+     * @return The next power of 2 or the value itself if it is a power of 2
+     */
+    private static int findNextPositivePowerOfTwo(final int value)
```

Comment:
Seems like this is a relatively generic method.  Can we make it more generally available?

Another area that may be able to take advantage: [PoolThreadCache](https://github.com/netty/netty/blob/master/buffer/src/main/java/io/netty/buffer/PoolThreadCache.java#L361).